pkg/kvstore/base_test.go

```diff
@@ -91,14 +93,50 @@ func (s *BaseTests) TestGetSet(c *C) {
 		c.Assert(err, IsNil)
 		c.Assert(val, IsNil)
 
-		val, err = GetPrefix(context.Background(), testKey(prefix, i))
+		key, val, err = GetPrefix(context.Background(), testKey(prefix, i))
 		c.Assert(err, IsNil)
 		c.Assert(val, IsNil)
+		c.Assert(key, Equals, "")
 	}
 
-	val, err = GetPrefix(context.Background(), prefix)
+	key, val, err = GetPrefix(context.Background(), prefix)
 	c.Assert(err, IsNil)
 	c.Assert(val, IsNil)
+	c.Assert(key, Equals, "")
+}
+
+func (s *BaseTests) TestGetPrefix(c *C) {
```

Comment:
NB: This new test doesn't actually test the bug, but it was also missing. I played around a little with a `pkg/kvstore/allocator` test to validate the fix but I couldn't figure it out just yet, will need more time to work through unit test structure issues.